python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -701,6 +702,45 @@ def cummax(self):
         """Get the column-wise cumulative maximum value in each group."""
         return self.agg("cummax")
 
+    def shift(self, periods=1, freq=None, axis=0, fill_value=None):
+        """
+        Shift each group by `periods` positions.
+
+        Parameter "freq" is unsupported.
+
+        Parameters
+        ----------
+        periods : int, default 1
+        Number of periods to shift.
+
+        freq : str, unsupported
+
+        axis : 0, axis to shift
+        Shift direction. Only row-wise shift is supported
+
+        fill_value : optional
+        The scalar value to use for newly introduced missing values. Should
+        match the dtype of columns to fill.
+
+        Returns
+        Series or DataFrame
+        Object shifted within each group.
+        """
```

Comment:
```suggestion
        """
        Shift each group by ``periods`` positions.

        Parameters
        ----------
        periods : int, default 1
            Number of periods to shift.
        freq : str, unsupported
        axis : 0, axis to shift
            Shift direction. Only row-wise shift is supported
        fill_value : optional
            The scalar value to use for newly introduced missing values. Should
            match the dtype of columns to fill.

        Returns
        -------
        Series or DataFrame
            Object shifted within each group.

        Notes
        -----
        Parameter ``freq`` is unsupported.
        """
```

The docstring will need to be formatted